### DISCUSSION TITLE
What architecture do you run Sandboxie on?

### DISCUSSION THREAD

**Discussion by bastik-1001** [sandboxie-plus/Sandboxie#2135] [Polls]
This poll attempts to answer a question I have been curious about, namely what architecture people run Sandboxie on, since it provides Windows binaries for 32bit and 64bit versions of Windows. Which leads to the question if the architectures do not allow users to switch from 32bit to 64bit.

**Comment by DavidXanatos** [comment]
Guess who voted ARM64 LOL
About the question the sandboxie architecture must match the OS architecture, otherwhere it cant work.
But I would also be very curious about how many users are still a 32 bit operating system.

**Reply by bastik-1001** [reply]
:)

I wondered the very same and guessed that it might be due to the architecture.

I myself, have a 64bit processor and 64bit OS, and still need to run some 32bit software. I knew that one can not use 64bit software on a 32bit machine. I was not aware that Sandboxie needed the OS version to match.

**Reply by DavidXanatos** [reply]
MSFT has implemented variouse WOW (windows on windows) mechanisms for user mode applications, while the kernel is always native. So for example a ARM64 kernel requires arm64 drivers and can not do anything with a ARM32 or x64 driver. 
So the sandboxie driver must always match the platform. 

With regard to usermode applications while let say on ARM64 they can be ARM64, ARM32, x64 or x86 the processes start their execution in an entry point of the native ntdll.dll only after some or some more initialization the execution transitions to whatever code the actual application was compiled to.
Sandboxie also intercepts this early initialization hence it must bring native low level injection code for this.
Only once the SbieDll.dll is loaded it is done so in the applications architecture.
Hence x64 sandboxie brings a x86 sandboxie dll,
and therefor the arm64 sandboxie (to be released soon) will bring a x86 and a x64 compatible (arm64ec) sandboxie.dll

BTW: 32 bit arm will not be supported as imho there is next to no software out there compiled that way anyways.
